pkg/controller/containerlaunchpriority/container_launch_priority.go

```diff
@@ -0,0 +1,217 @@
+// +kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+
+func (r *ReconcileContainerLaunchPriority) Reconcile(request reconcile.Request) (res reconcile.Result, err error) {
+	start := time.Now()
+	if klog.V(3) {
+		klog.Infof("Starting to process Pod %v", request.NamespacedName)
+	}
+	defer func() {
+		if err != nil {
+			klog.Warningf("Failed to process Pod %v, elapsedTime %v, error: %v", request.NamespacedName, time.Since(start), err)
+		} else {
+			klog.Infof("Finish to process Pod %v, elapsedTime %v", request.NamespacedName, time.Since(start))
+		}
+	}()
+
+	// get pod and barrier
+	var pod = &v1.Pod{}
+	err = r.Get(context.TODO(), request.NamespacedName, pod)
+	if errors.IsNotFound(err) {
+		return reconcile.Result{}, nil
+	} else if err != nil {
+		return reconcile.Result{}, err
+	}
+	var barrier = &v1.ConfigMap{}
+	var barrierNamespacedName = types.NamespacedName{
+		Namespace: pod.GetNamespace(),
+		Name:      pod.Name + "-barrier",
+	}
+	err = r.Get(context.TODO(), barrierNamespacedName, barrier)
+	if errors.IsNotFound(err) {
+		barrier.Namespace = pod.GetNamespace()
+		barrier.Name = pod.Name + "-barrier"
+		barrier.OwnerReferences = append(barrier.OwnerReferences, metav1.OwnerReference{
+			APIVersion: "v1",
+			Kind:       "Pod",
+			Name:       pod.Name,
+			UID:        pod.UID,
+		})
+		temErr := r.Client.Create(context.TODO(), barrier)
+		if temErr != nil {
+			return reconcile.Result{}, temErr
+		}
+	}
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// set next starting pod
+	if !r.getConditionStatus(pod, v1.ContainersReady) {
+		var patchKey = r.findNextPatchKey(pod)
+		key := "p_" + strconv.Itoa(patchKey)
+		if err = r.patchOnKeyNotExist(barrier, key); err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileContainerLaunchPriority) validate(pod *v1.Pod) bool {
+	if len(pod.Spec.Containers) == 0 {
+		return false
+	}
+	for _, v := range pod.Spec.Containers[0].Env {
+		if v.Name == priorityBarrier {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *ReconcileContainerLaunchPriority) getConditionStatus(pod *v1.Pod, conditionType v1.PodConditionType) bool {
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == conditionType {
+			return condition.Status == v1.ConditionTrue
+		}
+	}
+	return false
+}
+
+func (r *ReconcileContainerLaunchPriority) findNextPatchKey(pod *v1.Pod) int {
+	var priority = make([]int, 0)
+	var containerPendingSet = make(map[string]bool)
+	for _, status := range pod.Status.ContainerStatuses {
+		if status.Ready {
+			continue
+		}
+		containerPendingSet[status.Name] = true
+	}
+	for _, c := range pod.Spec.Containers {
+		if _, ok := containerPendingSet[c.Name]; ok {
+			for _, e := range c.Env {
+				if e.Name == priorityBarrier {
+					p, _ := strconv.Atoi(e.ValueFrom.ConfigMapKeyRef.Key[2:])
```

Comment:
plz wrap the parse logic in a separate function (such as getLauchPriority), and avoid hard code magic number like "2"